src/crimson/net/SocketConnection.cc

```diff
@@ -801,8 +801,7 @@ seastar::future<> SocketConnection::client_handshake(entity_type_t peer_type,
       validate_peer_addr(saddr, peer_addr);
 
       if (my_addr != caddr) {
-        // take peer's address for me, but preserve my port/nonce
-        caddr.set_port(my_addr.get_port());
+        // take peer's address for me, but preserve my nonce
```

Comment:
why would you want to learn the port from the peer? i understand that the client side's port is always 0 here, but it does not hurt, right? also, `SimpleMesseger::learned_addr()` also preserves the nonce *and* port.

i am fine either way as long as it does not break the messenger.

Reply:
It will be clearer for debugging/logging purposes if client doesn't drop this port information. And I'm testing multiple-connection scenario now...

Currently `AsyncMessenger::learned_addr()` may learn the port from peer at certain condition. If this would break seastar-msgr, we can store it inside `SocketConnection` separately.